app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateAndEditRecipeForm/styledComponents';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+
+export default function({
+  recipeData,
+  setIsCreateAndEditRecipeFormOpen,
+  setSnackbar,
+  recipeFormAction,
+}: ICreateAndEditRecipeFormProps) {
+  const EditRecipeFormAction = 'editRecipe';
+  const [isNameError, setIsNameError] = useState(false);
+  const StyledLabel = getLabelStyle(isNameError);
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisable, setIsSaveDisable] = useState<boolean>(true);
+
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  useEffect(() => {
+    if (isNameError) {
+      setIsSaveDisable(true);
+    }
+  }, [isNameError]);
+
+  useEffect(() => {
+    setRecipeFormData(recipeData);
+  }, [recipeData]);
+
+  useEffect(() => {
+    if (
+      recipeFormData.recipeName === '' ||
+      recipeFormData.description === '' ||
+      recipeFormData.recipeName?.trim().length === 0 ||
+      recipeFormData.description?.trim().length === 0
+    ) {
+      setIsSaveDisable(true);
+    } else {
+      setIsNameError(false);
+      setIsSaveDisable(false);
+    }
+  }, [recipeFormData]);
+
+  const onCancel = () => {
+    setIsCreateAndEditRecipeFormOpen(false);
+    setIsNameError(false);
+  };
+
+  const onFormHandle = (event: FormEvent) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeData) => {
+    if (recipeFormAction === EditRecipeFormAction) {
+      const params = {
+        context: getCurrentNamespace(),
+        recipe_id: recipeData.recipeId.recipeId,
+      };
+      const payload = {
+        recipeName: recipeFormData.recipeName,
+        description: recipeFormData.description,
+        directives: recipeFormData.directives,
+      };
+      MyDataPrepApi.updateRecipe(params, payload).subscribe(
+        () => {
+          setIsNameError(false);
+          setIsCreateAndEditRecipeFormOpen(false);
+          setSnackbar({
+            open: true,
+            isSuccess: true,
+            message: `${recipeSteps.length} ${T.translate(
+              'features.WranglerNewUI.RecipeForm.labels.recipeSaveSuccessMessage'
+            )}`,
+          });
+        },
+        (err) => {
+          if (err.response.message) {
+            setIsNameError(true);
+          } else {
+            setIsCreateAndEditRecipeFormOpen(true);
+            setSnackbar({
+              open: true,
+              isSuccess: false,
+              message: T.translate(
+                'features.WranglerNewUI.RecipeForm.labels.errorMessage'
+              ).toString(),
+            });
+          }
+        }
+      );
+    }
+  };
+
```

Comment:
`onChange` handlers for form fields should be defined as functions here. 

```js
function onRecipeNameChange(event: React.ChangeEvent<HTMLInputElement>): void {
  handleSetRecipeFormData({ 
    ...recipeFormData, 
    recipeName: event.target.value,
  });
}

function onRecipeDescriptionChange(event: React.ChangeEvent<HTMLInputElement>): void {
  handleSetRecipeFormData({ 
    ...recipeFormData, 
    description: event.target.value,
  });
}
``` 

See next comments.

Reply:
now using onChange handlers for form fields